doc/guides/modules.rst

```diff
@@ -253,9 +263,43 @@ Where 23 in the example above indicated the pull request number submitted to the
 *my_module* repository. Once the module changes are reviewed and merged, the
 revision needs to be changed to the commit hash from the module repository.
 
-
+Upmerging forks
+===============
+
+When a module is in fact a fork of an existing, external project that is hosted
+on GitHub or elsewhere, it is often necessary to upmerge it: update the copy
+that is hosted under the `zephyrproject-rtos`_ GitHub organization with the
+latest changes from the external upstream repository.
+
+The procedure for upmerging module repositories that are forks of external
+projects is constrained by the following two factors:
+
+* History must **never** be rewritten in any repository that is referenced in
+  the `west.yml`_ manifest file, since that could break older revisions of the
```

Comment:
Why? We could prevent the old commits being garbage collected using tags, as we do downstream.